code/game/objects/structures/stairs.dm

```diff
@@ -24,6 +24,10 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/stairs/multiz, 0)
 		force_open_above()
 		build_signal_listener()
 	update_surrounding()
+	var/turf/open/openspace/T = get_step_multiz(get_turf(src), UP)
+	if(T && istype(T, /turf/open/openspace))
+		SSminimaps.add_marker(T, MINIMAP_FLAG_ALL, image('icons/UI_icons/map_blips.dmi', null, "stairs_down", MINIMAP_LABELS_LAYER))
+	SSminimaps.add_marker(src, MINIMAP_FLAG_ALL, image('icons/UI_icons/map_blips.dmi', null, "stairs_up", MINIMAP_LABELS_LAYER))
 
```

Comment:
Won't this give down stairs both markers

Reply:
We don't really have down stairs exactly. We have stairs that point up and make a hole above them, you can enter that hole to go back down (but you knew that).

What this code is doing is assigning the map marker to the stairs structure itself, and then the assigning a down marker to the open space above (unless it doesn't exist). It's not the greatest implementation and I'm open to better suggestions.